src/SourceLocator/SourceStubber/PhpStormStubsSourceStubber.php

```diff
@@ -415,6 +445,57 @@ private function getAbsoluteFilePath(string $filePath): string
         return sprintf('%s/%s', $this->getStubsDirectory(), $filePath);
     }
 
+    /**
+     * @param list<Node\Name> $nameNodes
+     *
+     * @return list<Node\Name>
+     */
+    private function modifyExtendsImplementsByPhpVersion(string $className, array $nameNodes): array
+    {
+        $modifiedNames = [];
+        foreach ($nameNodes as $nameNode) {
+            $name = $nameNode->toString();
+
+            if ($className === ParseError::class) {
+                if ($name === CompileError::class && $this->phpVersion < 70300) {
+                    $modifiedNames[] = new Node\Name\FullyQualified(Error::class);
+                    continue;
+                }
+            } elseif ($className === SplFixedArray::class) {
+                if ($name === JsonSerializable::class && $this->phpVersion < 80100) {
+                    continue;
+                }
+
+                if ($name === IteratorAggregate::class && $this->phpVersion < 80000) {
+                    continue;
+                }
+
+                if ($name === Iterator::class && $this->phpVersion >= 80000) {
+                    continue;
+                }
+            } elseif ($className === SimpleXMLElement::class) {
+                if ($name === RecursiveIterator::class && $this->phpVersion < 80000) {
+                    continue;
+                }
+            } elseif (in_array($className, [DatePeriod::class, PDOStatement::class, true])) {
```

Comment:
IMO `===` in this case - removes `in_array()` and array instantiation (even from opcache interned values) each time